### PR TITLE
Javascipt js events

### PR DESCRIPTION
**Pull request by Laylo309**
- [x] Copy the files calculate.js and operate.js into a logic/ directory in your project.
- [x] Analyze the files calculate.js and operate.js.
- [x] Import the files in the Calculator component.
- [x] Implement the event handlers and used the math logic from calculate.js and operate.js in your React component.
- [x] Practice using state and props, pass props to child components, use lifecycle methods, lift the state up.
- [x] Test the app in the browser. It is able to perform math operations.

### CODE REVIEW

File: src/components/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+ */
+export default function calculate(obj, buttonName) {
+  if (buttonName === 'AC') {
+    return {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
```
Comment (DeliceLydia) [inline src/components/logic/calculate.js]: - [x] Consider displaying `zero` when you click on the `AC` button and it looks good when you see operations on your right side

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Laylo309, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/logic/calculate.js (hunk 1 above)
Comment (Laylo309) [reply src/components/logic/calculate.js]: Done ✅

**Review by ZawadiSumba66 — APPROVED** [review]
Hi  @Laylo309 👋 ,
## STATUS APPROVED ✔️  
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**